### DISCUSSION TITLE
Cannot use findByID after DB restore

### DISCUSSION THREAD

**Discussion by dacgray** [payloadcms/payload#6699] [Q&A]
### Describe the Bug

We want to generate dev seed data from our production database.  We need to remove sensitive data, so in prod we iterate through our collections and generate safe seed JSON files.

EG:

users.json
```json
[
  {
    "_id": "6516875ee6d9097e0e0d6d35",
    "name": "Foo",
    "mfaKey": "",
    "role": "admin",
    "canTranslateLocales": [],
    "email": "foo@example.com",
    "salt": "",
    "hash": "",
    "loginAttempts": 0,
    "__v": 0,
    "lockUntil": null
  },
  {
    "_id": "6523b9b386ebc1443f5ec3c8",
    "name": "Bar",
    "mfaKey": "",
    "role": "admin",
    "canTranslateLocales": [],
    "email": "bar@example.com",
    "salt": "",
    "hash": "",
    "loginAttempts": 0,
    "__v": 0,
    "lockUntil": null,
    "resetPasswordExpiration": "2024-02-28T10:27:21.833Z",
    "resetPasswordToken": ""
  },
]
```

In dev we run a migration that populates the DB with these files using mongo insertMany().

```javascript
await payload.db.connection.collection('foo').insertMany(list)
```

The problem is that we can't fetch models by their ID.  

We can get data with `payload.find()`,  and in the UI we can see the collection list view, but we get "Nothing Found" in the model view.

We can fetch models by ID using the mongo client:

```javascript
// This works.
const mongoModel = await collection.findOne({_id: 'ABC123'})
```

But not with payload:

```javascript
// This doesn't work.
 const payloadModel = await payload.findByID({
    collection: 'foo',
    id: 'ABC123',
})
```

Our collections' schema are rather complex, which is why we want to generate dev seed data from prod.  We need to maintain relationships between collections, so we need to populate the dev DB with IDs.

I would very much like to avoid creating an ID field on all collections just for this.

Are we missing a step?

![image](https://github.com/payloadcms/payload/assets/12794616/2bb688de-b1b1-40bc-8abc-00da9c317167)

![image](https://github.com/payloadcms/payload/assets/12794616/0d81f7cc-624a-4c51-8479-f27ff129a8e7)

### To Reproduce

As above

### Payload Version

2.19.3

### Adapters and Plugins

"@payloadcms/bundler-webpack": "^1.0.6",     "@payloadcms/db-mongodb": "^1.3.2",     "@payloadcms/plugin-cloud-storage": "^1.1.1",     "@payloadcms/plugin-sentry": "^0.0.6",     "@payloadcms/richtext-slate": "^1.3.1",

**Comment by jmikrut** [comment]
Hey @dacgray — do you have versions / drafts enabled on these collections? You need to also _restore_ the corresponding versions collection if so. 

If that's not it, then you might need to create ObjectIds out of the `_id` property before you go to insert them. 

Once your documents are stored in your DB, you need to make sure that the `_id` shows up as an `ObjectId` and I'm guessing yours are just being inserted as strings.

Give that a shot!

**Reply by dacgray** [reply]
Creating ObjectIds worked.  Thank you.

For anyone who lands here:

Run `yarn add -D mongodb` to use `import {ObjectId} from 'mongodb';`

Insert using:

```
await payload.db.connection
    .collection(collectionName)
    .insertMany(
        list.map(data => ({
            ...data,
            _id: new ObjectId(data._id),
        }))
    )
```
